packages/just-scripts-utils/src/transform.ts

```diff
@@ -2,30 +2,126 @@ import glob from 'glob';
 import path from 'path';
 import fse from 'fs-extra';
 import handlebars from 'handlebars';
+import { logger } from './logger';
 
-export function transform(srcPath: string, destPath: string, data?: any) {
-  const templateFiles = [...new Set([...glob.sync('**/*', { cwd: srcPath }), ...glob.sync('**/.*', { cwd: srcPath })])];
+export interface ITransformResult {
+  /** Whether a fatal error occurred */
+  error?: boolean;
+  /** Number of files/folders processed successfully (not counting warnings) */
+  processed?: number;
+  /** Number of files/folders for which processing was unsuccessful */
+  warnings?: number;
+}
+
+/**
+ * Apply a template to a project directory. If the directory doesn't exist, it will be created.
+ * If the directory exists, contents will be merged with the files/folders from the template.
+ *
+ * @param templateDir Path to the directory containing the template
+ * @param projectDir Path to the destination project to create/update
+ * @param templateData If the template contains any handlebars files, this will be passed to them
+ * when compiling
+ * @returns Object indicating whether the transform was successful
+ */
+export function transform(
```

Comment:
Could we call this `applyTemplate` or something like that instead? Seems a bit clearer.